src/HouseofCat.RabbitMQ/Messages/Letter.cs

```diff
@@ -95,7 +95,7 @@ public Letter(string exchange, string routingKey, byte[] data, string id, byte p
         public Letter Clone()
         {
             var clone = this.Clone<Letter>();
-            clone.LetterMetadata = LetterMetadata.Clone<LetterMetadata>();
+            clone.LetterMetadata = (LetterMetadata)LetterMetadata.Clone<LetterMetadata>();
```

Comment:
We could change the extension method too for cleaner code:

public static T Clone<T>(this IMetadata metadata) where T: IMetadata, new()

Reply:
Oh the markup removed the T generic around the Clone, but you get the gist hopefully..

Reply:
Good point, keeping with the current trend.